src/tmiic.cpp

```diff
@@ -0,0 +1,173 @@
+//*****************************************************************************
+// Filename   : tmiic.cpp                   Namespace: tmiic                                   
+//
+// Author     : Franck SIMON                Creation date: 07 may 2020 
+//
+// Description: Store functions for temporal mode of miic (tmiic)
+//
+// Changes history:
+// - 07 may 2020 : initial version
+//*****************************************************************************
+
+//=============================================================================
+// INCLUDE SECTION
+//=============================================================================
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <vector>
+
+#include "environment.h"
+#include "structure.h"
+#include "tmiic.h"
+
+//=============================================================================
+// CONSTANTS
+//=============================================================================
+
+//=============================================================================
+// NAMESPACES
+//=============================================================================
+namespace tmiic {
+
+using std::string;
+using namespace miic::structure;
+using namespace miic::utility;
+
+//=============================================================================
+// FUNCTIONS
+//=============================================================================
+// countNbNodesNotLagged
+//-----------------------------------------------------------------------------
+// Description: Counts the number of nodes not lagged
+//
+// Detail: Counts the number of nodes ending with lag0
+//
+// Params: 
+// - Environment&: the environment structure
+//
+// Returns: int, the number of nodes not lagged
+//-----------------------------------------------------------------------------
+int countNbNodesNotLagged (Environment& environment) {
```

Comment:
This number could effectively be sent from R to C++ part as an extra parameter. As discussed, IMHO, adding more and more parameters every time we have an information in the R part usefull to the C++ is not good: it will make the inferface more complex whilst if should be IMO kept minimal.

Reply:
Firstly, I agree with you that the interface should be kept minimal, but that doesn't mean we should avoid adding parameters at all cost. The parameter passage is refactored to use a `arg_list` specifically for its flexibility, which allow us to pass extra parameters without modifying the interface every time.

Now what you have traded for a minimal interface is some extra functions. To compute something we already know is worse than having one extra parameter, as it brings down the overall readability and simplicity of the code.